src/picongpu/include/fields/background/templates/bgrTWTS.def

```diff
@@ -0,0 +1,263 @@
+                        const float_X phi               = 90.*(PI/180.),
+                        const float_X beta_0            = 1.0,
+                        const float_64 tdelay_user_SI   = 0.0,
+                        const bool auto_tdelay          = true );
+            
+            
+            /** Specify your background field B(r,t) here
+             *
+             * \param cellIdx The total cell id counted from the start at t=0
+             * \param currentStep The current time step */
+            HDINLINE float3_X
+            operator()( const DataSpace<simDim>& cellIdx,
+                        const uint32_t currentStep ) const;
+
+            /** Calculate the By(r,t) field here
+             *
+             * \param pos Spatial position of the target field.
+             * \param time Absolute time (SI, including all offsets and transformations)
+             *  for calculating the field */
+            HDINLINE float_T
+            calcTWTSBy( const float3_64& pos, const float_64 time ) const;
+
+            /** Calculate the Bz(r,t) field here
+             *
+             * \param pos Spatial position of the target field.
+             * \param time Absolute time (SI, including all offsets and transformations)
+             *  for calculating the field */
+            HDINLINE float_T
+            calcTWTSBz( const float3_64& pos, const float_64 time ) const;
+
+            /** Calculate the SI position vectors that later enter
+             *  the By(r, t) and Bz(r, t) calculations as r.
+             * \tparam T_dim Specializes for the simulation dimension
+             * \param cellIdx The total cell id counted from the start at timestep 0. */
+            template<unsigned T_dim>
+            HDINLINE PMacc::math::Vector<float3_64,numComponents>
+            getBfieldPositions_SI(const DataSpace<simDim>& cellIdx) const;
+            
+            /** Calculate the B-field vector of the TWTS laser in SI units.
+             * \tparam T_dim Specializes for the simulation dimension
+             * \param cellIdx The total cell id counted from the start at timestep 0
+             * \return B-field vector of the rotated TWTS field in SI units */
+            template<unsigned T_dim>
+            HDINLINE float3_X
+            getTWTSBfield_Normalized(
+                            const PMacc::math::Vector<float3_64,numComponents>& eFieldPositions_SI,
+                            const float_64 time) const;
+            
+            /** Obtain the SI time that later enters the By(r, t) and Bz(r, t) calculations as r.
+             * \tparam T_dim Specializes for the simulation dimension
+             *  \param CurrentStep current time step number of simulation
+             *  \return time in SI units */
+            template<unsigned T_dim>
+            HDINLINE float_64 getTime_SI(const uint32_t currentStep) const;
+
+            HDINLINE PMacc::math::Vector<float3_X,numComponents>
```

Comment:
all `dummyCast`s are defined and implemented in `TWTSFieldE` and `TWTSFieldB`. As I can see the implementations were equal. Thus it would be nice to move the function out of the classes and only implement them once.